src/utils/matmul_helper.h

```diff
@@ -492,11 +492,15 @@ class MMHelper {
 
         // W8A8
         else if constexpr (std::is_same_v<WeiT, w8a8_t>) {
-            weight.Resize(K, N);
             auto tag = trans ? dnnl::memory::format_tag::ba : dnnl::memory::format_tag::ab;
             dnnl::memory B_mem({{K, N}, dnnl::memory::data_type::s8, tag}, get_dnnl_engine(), src.Data());
-            dnnl::memory packedB_mem({{K, N}, dnnl::memory::data_type::s8, get_amx_s8s8s32_weight_data_type()},
-                    get_dnnl_engine(), weight.Data());
+            dnnl::memory::desc desc {{K, N}, dnnl::memory::data_type::s8, get_amx_s8s8s32_weight_data_type()};
+            // When converting to oneDNN blocked memory format, padded dims can be larger than [K, N]
+            // Resize down Matrix does not change underlying buffer.
```

Comment:
@xiangzez pls add a todo here (better to add some reserve like function in Matrix, as current 2 times Resize has risks in cases ...)